odoo/models.py

```diff
@@ -3002,6 +3002,14 @@ def _update_field_translations(self, field_name, translations, digest=None):
         """
         self.ensure_one()
 
+        installed_lang = list(code for code, _ in self.env['res.lang'].get_installed())
+        missing = set(translations) - set(installed_lang)
+        if missing:
+            raise UserError(
+                _("The following language is not activated: %(missing_names)s",
+                missing_names=', '.join(missing))
+            )
```

Comment:
some suggestions:
1. you can at least remove the `list` and keep it as a generator
2. maybe it would be better to have constant names for the uninstalled/missing/activated languages
```py
         uninstalled_langs = set(translations) - set(installed_langs)
         if uninstalled_langs:
             raise UserError(
                 _("The following languages are not installed: %s",
                 ', '.join(uninstalled_langs))
             )
```

Reply:
I think they have no difference, since we allow users to create a user defined language, e.g. language xx_YY
Inactivate languages are the same as non-existing languages
But I am also OK with the `missing`